### PR TITLE
Cherry pick #4296 into G.0 release-0.28 branch

### PR DESCRIPTION
**Pull request by harisund**
### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

**Review by zjs — APPROVED** [review]

**Comment by zjs** [comment]
IMO, this is good to merge once #4296 is merged

**Review by imikushin — APPROVED** [review]

**Review by lubronzhan — APPROVED** [review]

**Pull request merged**
